src/Service/Program.cs

```diff
@@ -119,6 +119,18 @@ public static ILoggerFactory GetLoggerFactoryForLogLevel(LogLevel logLevel)
                     builder.AddFilter(category: "Microsoft", logLevel);
                     builder.AddFilter(category: "Azure", logLevel);
                     builder.AddFilter(category: "Default", logLevel);
+
+                    // For Sending all the ILogger logs to Application Insights
+                    if (Startup._applicationInsightsOptions.Enabled)
+                    {
+                        builder.AddApplicationInsights(configureTelemetryConfiguration: (config) =>
+                            config.ConnectionString = Startup._applicationInsightsOptions.ConnectionString,
+                            configureApplicationInsightsLoggerOptions: (options) => { }
+                        )
+                        .AddFilter<ApplicationInsightsLoggerProvider>("", logLevel)
+                        .AddProvider(new ApplicationInsightsLoggerProvider(Startup._telemetryClient!));
```

Comment:
Would ILogger be a valid provider here for appinsights?

Reply:
ApplicationInsightsLoggerProvider is based on top of ILoggerProvider itself, we need to customize it for capturing errors and creating event for Error case.